src/main/java/org/broadinstitute/dsm/jobs/PubSubLookUpJob.java

```diff
@@ -0,0 +1,92 @@
+        String projectId = "hl7-dev-274911";
+        String subscriptionId = "pegah-results-sub";
+
+        try{
+            ProjectSubscriptionName subscriptionName =
+                    ProjectSubscriptionName.of(projectId, subscriptionId);
+
+            // Instantiate an asynchronous message receiver.
+            MessageReceiver receiver =
+                    (PubsubMessage message, AckReplyConsumer consumer) -> {
+                        // Handle incoming message, then ack the received message.
+                        System.out.println("Id: " + message.getMessageId());
+                        System.out.println("Data: " + message.getData().toStringUtf8());
+                        writeResultsIntoDB(message);
+                        consumer.ack();
+                    };
+
+            Subscriber subscriber = null;
+            try {
+                subscriber = Subscriber.newBuilder(subscriptionName, receiver).build();
+                // Start the subscriber.
+                subscriber.startAsync().awaitRunning();
+                logger.info("Listening for messages on %s:\n", subscriptionName.toString());
+                // Allow the subscriber to run for 30s unless an unrecoverable error occurs.
+                subscriber.awaitTerminated(1L, TimeUnit.MINUTES);
+            } catch (TimeoutException timeoutException) {
+                // Shut down the subscriber after 30s. Stop receiving messages.
+                subscriber.stopAsync();
+            }
+        }
+        catch (Exception e){
+            throw new RuntimeException("Failed to get results from pubsub ", e);
+        }
+
+    }
+
+    public static void writeResultsIntoDB(PubsubMessage message){
+        String data = message.getData().toStringUtf8();
+        TestBostonResult testBostonResult = new Gson().fromJson(data, TestBostonResult.class);
+        String query = "UPDATE ddp_kit SET  test_result = ?, result_date= ?  WHERE dsm_kit_id <> 0 and  dsm_kit_id  in ( select  dsm_kit_id from (select * from ddp_kit) as something where kit_label= ?  )";
+        TestBostonResult finalTestBostonResult = testBostonResult;
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                if (StringUtils.isNotBlank(finalTestBostonResult.result) && StringUtils.isNotBlank(finalTestBostonResult.date) && StringUtils.isNotBlank(finalTestBostonResult.sampleId)) {
+                    stmt.setString(1, finalTestBostonResult.result);
+                    stmt.setString(2, finalTestBostonResult.date);
+                    stmt.setString(3, finalTestBostonResult.sampleId);
+                    stmt.executeUpdate();
+                }
+            }
+            catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            return dbVals;
+        });
+
```

Comment:
Can we log cases where we are *changing* the result (instead of going from zero results  to one result)?

Reply:
A little confused here, do you mean not change them, just log instead?

Reply:
We should update the value and log that it has been updated.

Reply:
👍